api/gdpr.py

```diff
@@ -0,0 +1,165 @@
+                    {
+                        "organisation": [
+                            app.organisation.name,
+                            app.organisation.identifier,
+                            app.organisation.email,
+                            app.organisation.core_business,
+                            app.organisation.core_business_fi,
+                            app.organisation.core_business_en,
+                            app.organisation.core_business_sv,
+                        ]
+                    }
+                )
+            if app.organisation and app.organisation.address:
+                applications.append(
+                    {
+                        "organisation_address": [
+                            app.organisation.address.post_code,
+                            app.organisation.address.street_address,
+                            app.organisation.address.street_address_fi,
+                            app.organisation.address.street_address_en,
+                            app.organisation.address.street_address_sv,
+                            app.organisation.address.city,
+                            app.organisation.address.city_fi,
+                            app.organisation.address.city_en,
+                            app.organisation.address.city_sv,
+                        ]
+                    }
+                )
+            if app.billing_address:
+                applications.append(
+                    {
+                        "billing_address": [
+                            app.billing_address.post_code,
+                            app.billing_address.street_address,
+                            app.billing_address.street_address_fi,
+                            app.billing_address.street_address_en,
+                            app.billing_address.street_address_sv,
+                            app.billing_address.city,
+                            app.billing_address.city_fi,
+                            app.billing_address.city_en,
+                            app.billing_address.city_sv,
+                        ]
+                    }
+                )
+
+        return applications
+
+    @property
+    def email(self):
+        return self.user.email
+
+
+class TilavarauspalveluGDPRAPIView(GDPRAPIView):
+    def get_object(self) -> User:
+        user = get_object_or_404(User, uuid=self.kwargs["uuid"])
+        profile = ProfileUser(user)
+        # Permissio check???
+        self.check_object_permissions(self.request, profile)
+        return profile
+
+    def _delete(self):
+        profile_user = self.get_object()
+        if not self._check_user_can_be_anonymized(profile_user.user):
+            raise DeletionNotAllowed()
+        anonymize_user_data(profile_user.user)
+
+    def _check_user_can_be_anonymized(self, user):
+        # TODO: Something to check with applications, are there any applications that needs to be handled still.
```

Comment:
Is this method complete? Will this be implemented at a later stage? :thinking:

Reply:
yes, it is reminder for later if we would like to do some checks that if the user data is not anonymizable => not yet specsed